scRNA-seq-advanced/02-celltype_assignment.Rmd

```diff
@@ -0,0 +1,465 @@
+    type = "jaccard",
+    cluster.fun = "louvain"
+  )
+)
+
+# add clusters to colData
+sce$nn_cluster <- nn_clusters
+```
+
+```{r}
+scater::plotUMAP(sce, 
+                 colour_by = "nn_cluster")
+```
+But what are these clusters? Do they correspond to cell types?
+
+## Cell surface markers
+
+The markers we have were presumably chosen for the fact that they mark particular cell types, so let's look at a few.
+
+CD3 is a marker of T-cells 
+
+```{r}
+scater::plotUMAP(sce, 
+                 colour_by = "CD3")
+```
+
+CD4+ T-cells
+
+```{r}
+scater::plotUMAP(sce, 
+                 colour_by = "CD4")
+```
+CD8+ T-cells
+
+```{r}
+scater::plotUMAP(sce, 
+                 colour_by = "CD8")
+```
+
+Looking at all of the markers here, but we probably don't want to do this, as we haven't introduced `purrr` yet.
+```{r}
+all_plots <- adt_tags %>%
+  purrr::map(~ scater::plotUMAP(sce, colour_by = .))
+
+patchwork::wrap_plots(all_plots)
+```
+
+
+So one thing we could do is construct a set of rules based on ADT tags, similar to how you might set gating rules for cell sorting.
+
+
+
+What thresholds should we use?
+If this is a useful marker, we should see a bimodal plot of the marked density.
+
+We'll do a bit of wrangling to pull out the ADT data, transpose it so each row is a cell,
+then convert it into a data frame for plotting. 
+(We have to first convert to a "regular" R matrix or R won't know how to convert it to a data frame)
+
+```{r}
+# convert to a data frame
+adt_df <- logcounts(altExp(sce)) %>%
+  t() %>% 
+  as.matrix() %>%
+  as.data.frame()
+```
+Now we can make a density plot with `ggplot2`!
+
+```{r}
+ggplot(adt_df, aes(x = CD3)) + 
+  geom_density() +
+  theme_bw()
+```
+
+Looking at a few of these, we can define some rough cutoffs.
+This is very manual and not particularly principled, but it _is_ a potential way to assign cell types.
+
+```{r}
+adt_df <- adt_df %>%
+  dplyr::mutate(
+    celltype = dplyr::case_when(
+      CD3 > 6.7 & CD4 > 8 ~ "CD4+ T cell",
+      CD3 > 6.7 & CD8 > 6 ~ "CD8+ T cell",
+      CD3 > 6.7 ~ "T cell"
+    )
+  )
+```
+
+```{r}
+sce$threshold_celltype <- adt_df$celltype
+scater::plotUMAP(sce, 
+                 colour_by = "threshold_celltype")
+```
+
+We could do similar things for any marker gene from the expression data as well.
+
+A more careful version of this is `AUCell`... do we want to introduce this?
+
+
+## Cell type annotation with `SingleR`
+
+Get some cell data
+```{r}
+# Don't ask about creating the cache directories, just create them
+ExperimentHub::setExperimentHubOption("ASK", FALSE)
+AnnotationHub::setAnnotationHubOption("ASK", FALSE)
+
+# Get Data from the Blueprint & Encode projects from celldex
+blueprint_encode <- celldex::BlueprintEncodeData(ensembl = TRUE)
+```
+What is this `blueprint_encode`?
+
+```{r}
+blueprint_encode
+```
+
+A SummarizedExperiment is very similar to a SingleCellExperiment, except rather than having each column as a cell, each column is a sample.
```

Comment:
In fact the SingleCellExperiment object structure is based on SummarizedExperiment, might be worth saying

Reply:
I planned to say that, but I guess it didn't make it in!